matcher/cache/cache.go

```diff
@@ -316,12 +313,12 @@ func (c *cache) setWithLock(
 	// a new cutover time and the old cached results are now invalid, therefore it's
 	// preferrable to invalidate everything to save the overhead of multiple invalidations.
 	if invalidate {
-		results = c.invalidateWithLock(nsHash, idHash, results)
+		results = c.invalidateWithLock(namespace, id, results)
 	}
 	res := results.source.ForwardMatch(id, fromNanos, toNanos)
-	newElem := &element{nsHash: nsHash, idHash: idHash, result: res}
+	newElem := &element{namespace: namespace, id: id, result: res}
```

Comment:
Can we make a copy of `namespace` and `id` here? We used to store hashes which was a value type so not copying was fine, now if we are storing byte slices, seems safer to copy them.

Reply:
Makes sense, will do.